pkg/gitserver/controller/factory/factory.go

```diff
@@ -0,0 +1,110 @@
+package factory
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/golang/glog"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/runtime"
+	kutil "k8s.io/kubernetes/pkg/util"
+	"k8s.io/kubernetes/pkg/watch"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	osclient "github.com/openshift/origin/pkg/client"
+	controller "github.com/openshift/origin/pkg/controller"
+	bccontroller "github.com/openshift/origin/pkg/gitserver/controller"
+)
+
+type BuildConfigControllerFactory struct {
+	HookDirectory string
+	PublicURL     *url.URL
+	Home          string
+	GitBinary     string
+	Namespace     string
+	Client        osclient.Interface
+	// Stop may be set to allow controllers created by this factory to be terminated.
+	Stop <-chan struct{}
+}
+
+// Create creates a new ConfigChangeController which is used to trigger builds on creation
+func (f *BuildConfigControllerFactory) Create() controller.RunnableController {
+	queue := cache.NewDeltaFIFO(cache.MetaNamespaceKeyFunc, nil, &keyListerGetter{})
```

Comment:
If you're using a DeltaFIFO you should be providing the list of the things that the git controller knows about, otherwise there's no cleanup that happens on your side.